### PR TITLE
Add documentation for DB2 client

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

**Comment by aguibert** [comment]
CC @vietj

**Comment by aguibert** [comment]
CI is good except for one unrelated failure

**Comment by BillyYccc** [comment]
You can execute `mvn package -Pdocs -DskipTests` and check the generated doc content in the target directory.

**Comment by aguibert** [comment]
For issue #654

File: vertx-db2-client/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+* Batch and cursor
+* Row streaming
+* RxJava 1 and RxJava 2
+* Direct memory to object without unnecessary copies
+* Java 8 Date and Time
+* HTTP/1.x CONNECT, SOCKS4a or SOCKS5 proxy support
+
+*Current limitations*
+
+* No TLS/SSL support
```
Comment (tsegismont) [inline vertx-db2-client/src/main/asciidoc/index.adoc:26]: Is this still true @aguibert ?

Comment (aguibert) [reply vertx-db2-client/src/main/asciidoc/index.adoc:26]: it was correct at the time of merging the PR, and then later on I added SSL/TLS support and updated the doc accordingly:
https://github.com/eclipse-vertx/vertx-sql-client/commit/b6e4b6ce1d7bde529242234ddf876ce5a80ad13d

Comment (tsegismont) [reply vertx-db2-client/src/main/asciidoc/index.adoc:26]: Great, thanks!

**Pull request merged**